File: day01/day01p1.py
import sys

left = []
right = []

for line in sys.stdin:
    n1, n2 = line.split()
    left.append(int(n1))
    right.append(int(n2))

left.sort()
right.sort()
print(sum(abs(n1 - n2) for n1, n2 in zip(left, right)))

File: day01/Day01P1.kt
import kotlin.math.abs

fun main() {
  val left = mutableListOf<Int>()
  val right = mutableListOf<Int>()
  for (line in generateSequence(::readLine)) {
    val (n1, n2) = line.split("""\s+""".toRegex()).map{ it.toInt() }
    left.add(n1)
    right.add(n2)
  }
  left.sort()
  right.sort()
  val result = left.zip(right).map { (n1, n2) -> abs(n1-n2) }.sum()
  println(result)
}

